### PR TITLE
copr: fix paging stop early unexpectedly with agg executors

### LINKED ISSUE #14291: copr: paging request stop unexpectedly
## Bug Report

Find by pingcap/tidb#41695

### What version of TiKV are you using?

```
TiKV 
Release Version:   6.7.0-alpha
Edition:           Community
Git Commit Hash:   5f5bb766ea056f1eb1320e32364084818a3eca64
Git Commit Branch: heads/refs/tags/v6.7.0-alpha
UTC Build Time:    2023-02-27 03:33:34
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
```

### Steps to reproduce

The TiDB issue mentioned a reproduce using TPCC check, here is a simple version, prepare data with this [script](https://gist.github.com/you06/a68c73d8e1e2f9016ba3f1262e3acbc8)

Run the query `select c1, c2, count(*) from t group by c1, c2 order by c1, c2`.

### What did you expect?

10240 rows.

### What did happened?

512 rows.

### PR DESCRIPTION
**Pull request by you06**
### What is changed and how it works?

Issue Number: Close #14291

What's Changed:

```commit-message
#14209 stop paging when the result set is drained. But when supporting agg paging,
the executors will also return drained when there is enough data returned.
This PR label the drain reasons and stop following paging when the resultset is really drained.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

TPCC check passed.

image

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by you06** [comment]
@tiancaiamao PTAL.

**Review by wshwsh12 — APPROVED** [review]

**Review by gengliqi — APPROVED** [review]

File: components/tidb_query_executors/src/interface.rs [hunk 1]
```diff
@@ -174,17 +174,42 @@ pub struct BatchExecuteResult {
...
+
+impl BatchExecIsDrain {
+    #[inline]
+    pub fn is_remain(&self) -> bool {
+        *self == BatchExecIsDrain::Remain
+    }
+
+    #[inline]
+    pub fn is_drain(&self) -> bool {
+        !self.is_remain()
```
Comment (gengliqi) [inline components/tidb_query_executors/src/interface.rs:213]: So if it's `PagingDrain`, `is_drain` return true?

Comment (you06) [reply components/tidb_query_executors/src/interface.rs:213]: May just use `!BatchExecIsDrain.is_remain()` as the stop scanning flag?

**Pull request merged**